framework/test/gtest/ServiceHooksTest.cpp

```diff
@@ -272,77 +267,44 @@ class TestServiceListenerHookFailure : public ServiceListenerHook
-                                                      hookProps1);
-
-  auto serviceEventListenerHook2 =
-    std::make_shared<TestServiceEventListenerHook>(2, context);
-  ServiceProperties hookProps2;
-  hookProps2[Constants::SERVICE_RANKING] = 0;
-  ServiceRegistration<ServiceEventListenerHook> eventListenerHookReg2 =
-    context.RegisterService<ServiceEventListenerHook>(serviceEventListenerHook2,
-                                                      hookProps2);
-
-  std::vector<int> expectedOrdering;
-  expectedOrdering.push_back(1);
-  expectedOrdering.push_back(1);
-  expectedOrdering.push_back(2);
-  US_TEST_CONDITION(serviceEventListenerHook1->ordering == expectedOrdering,
-                    "Event listener hook call order");
+protected:
+  Framework framework;
 
-  US_TEST_CONDITION(serviceListener1.events.empty(),
-                    "service event of service event listener hook");
-  US_TEST_CONDITION(serviceListener2.events.empty(),
-                    "no service event for filtered listener");
-
-  auto bundle = testing::InstallLib(context, "TestBundleA");
-  US_TEST_CONDITION_REQUIRED(bundle, "non-null installed bundle");
-  bundle.Start();
-
-  expectedOrdering.push_back(1);
-  expectedOrdering.push_back(2);
-  US_TEST_CONDITION(serviceEventListenerHook1->ordering == expectedOrdering,
-                    "Event listener hook call order");
+public:
+  ServiceHooksTest()
+    : framework(FrameworkFactory().NewFramework()){};
 
-  bundle.Stop();
+  ~ServiceHooksTest() override = default;
 
-  US_TEST_CONDITION(serviceListener1.events.empty(),
-                    "no service event due to service event listener hook");
-  US_TEST_CONDITION(serviceListener2.events.empty(),
-                    "no service event for filtered listener due to service "
-                    "event listener hook");
+  void SetUp() override
+  {
+    framework.Start();
+    auto bundles = framework.GetBundleContext().GetBundles();
+    ASSERT_FALSE(bundles.empty());
+    for (auto& b : bundles) {
+      if (b.GetSymbolicName() == "main") {
+        b.Start();
+        break;
+      }
+    }
+  }
 
-  eventListenerHookReg2.Unregister();
-  eventListenerHookReg1.Unregister();
+  void TearDown() override
+  {
+    framework.Stop();
+    framework.WaitForStop(std::chrono::milliseconds::zero());
+  }
+};
 
-  context.RemoveServiceListener(&serviceListener1,
-                                &TestServiceListener::ServiceChanged);
-  context.RemoveServiceListener(&serviceListener2,
-                                &TestServiceListener::ServiceChanged);
-}
+} // end unnamed namespace
 
-void TestListenerHook(const Framework& framework)
+TEST_F(ServiceHooksTest, TestListenerHook)
 {
   auto context =
-    testing::GetBundle("main", framework.GetBundleContext()).GetBundleContext();
+    cppmicroservices::testing::GetBundle("main", framework.GetBundleContext())
+      .GetBundleContext();
```

Comment:
this looks to be common for a number of tests in the `ServiceHooksTest` fixture. considering moving to a new fixture method for general use.